src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java

```diff
@@ -82,77 +66,52 @@ public DynamicACLFilter(final PluginSettings settings,
+     * a user and their roles.  We then dynamically update the ACL document, seed dashboards,
+     * and modify the kibana uri if needed.   
+     */
     public RestHandler wrap(final RestHandler original, final UnaryOperator<RestHandler> unaryOperator) {
         return new RestHandler() {
+            
+            private final RestHandler localHandler = new RestHandler() {
+                
+                @Override
+                public void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+                    if ((request = continueProcessing(request, channel)) != null) {
+                        original.handleRequest(request, channel, client);
+                    }
+                }
+            };
 
             @Override
             public void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
-                if ((request = continueProcessing(request, channel, client)) != null) {
-                    RestHandler handler = unaryOperator.apply(original);
-                    handler.handleRequest(request, channel, client);
-                }
+                
+                RestHandler handler = unaryOperator.apply(localHandler);
+                handler.handleRequest(request, channel, client);
             }
         };
     }
 
-    public RestRequest continueProcessing(RestRequest request, RestChannel channel, NodeClient client)
-            throws Exception {
+    public RestRequest continueProcessing(RestRequest request, RestChannel channel) throws Exception {
         try {
-            OpenshiftRequestContext requestContext = OpenshiftRequestContext.EMPTY;
-            try (StoredContext threadContext = threadPool.getThreadContext().stashContext()) {
-                if (enabled) {
-                    // create authenticates the request - if it returns null, this means
-                    // this plugin cannot handle this request, and should pass it to the
-                    // next plugin for processing e.g. client cert auth with no username/password
-                    // if create throws an exception, it means there was an issue with the token
-                    // and username and the request failed authentication
-                    requestContext = contextFactory.create(request);
-                    request = utils.modifyRequest(request, requestContext, channel);
-                    if (requestContext != OpenshiftRequestContext.EMPTY) {
-                        logRequest(request);
-                        final String kbnVersion = getKibanaVersion(request);
-                        kibanaSeed.setDashboards(requestContext, client, kbnVersion, cdmProjectPrefix);
-                        aclManager.syncAcl(requestContext);
-                    }
-
+            if (enabled && threadContext.getTransient(OPENSHIFT_REQUEST_CONTEXT) != null) {
```

Comment:
side question, but do we really need to check for `enabled` anymore? should we just deprecate that feature?

Reply:
I have wondered that for a while now and fully onboard with removing this all together.  I'm not sure how or why we would enable the feature.  It would be better if it were possible to hot reload it and maybe we would see an advantage for diagnosis or development when things are disabled.  I'll push changes to remove it.  maybe in a separate pr